### PR TITLE
Correct the iCal export link for the current month.

### PR DESCRIPTION
**Pull request by Camwyn**
The function that handles the link uses getLink() from Tribe__Events__Main.
For the current month, pretty permalinks will contain "now" for the month.

This replaces that "now" with the current month in 'Y-m' format for the iCal link only.

:camera: http://p.tri.be/dmNwm4

[TEC-3575]

[TEC-3575]: https://moderntribe.atlassian.net/browse/TEC-3575

### CODE REVIEW

File: src/Tribe/iCal.php [hunk 1]
```diff
@@ -84,8 +84,15 @@ public function get_ical_link( $type = 'home' ) {
...
 	public function month_view_ical_link( $event_date = null ) {
 		$tec = Tribe__Events__Main::instance();
+		$url = $tec->getLink( 'month', $event_date );
 
-		return add_query_arg( [ 'ical' => 1 ], $tec->getLink( 'month', $event_date ) );
+		/*
+		 * If the url contains "now" for the date, replace it with the current year-month.
+		 * Make sure we don't replace if it is part of another word - check for slashes.
+		 */
+		$url = str_replace( '/now/', '/' . date( 'Y-m' ) . '/', $url );
```
Comment (tr1b0t) [inline src/Tribe/iCal.php:93]: :no_entry_sign: date() is affected by runtime timezone changes which can cause date/time to be incorrectly displayed. Use gmdate() instead.

<sub>hash: 5201e70188e510944596fc849c07fab7</sub>

File: src/Tribe/iCal.php [hunk 2]
```diff
@@ -84,8 +84,15 @@ public function get_ical_link( $type = 'home' ) {
...
 	public function month_view_ical_link( $event_date = null ) {
 		$tec = Tribe__Events__Main::instance();
+		$url = $tec->getLink( 'month', $event_date );
 
-		return add_query_arg( [ 'ical' => 1 ], $tec->getLink( 'month', $event_date ) );
+		/*
+		 * If the url contains "now" for the date, replace it with the current year-month.
+		 * Make sure we don't replace if it is part of another word - check for slashes.
+		 */
+		$url = str_replace( '/now/', '/' . gmdate( 'Y-m' ) . '/', $url );
```
Comment (lucatume) [inline src/Tribe/iCal.php:93]: Do not use any system-based date handling (`date`, `gmdate`); use `Tribe__Date__Utils::build_date_object` instead.

Comment (lucatume) [reply src/Tribe/iCal.php:93]: I would have a more precise targeting of the `now` string:

1. There could not be a trailing `/`.
2. "now" is the English version of this, what about localized strings?

**Review by lucatume — CHANGES_REQUESTED** [review]
Add tests, please.

**Review by lucatume — APPROVED** [review]

**Pull request merged**